superset/db_engine_specs/mssql.py

```diff
@@ -85,6 +84,10 @@ def get_sqla_column_type(cls, type_: str) -> Optional[TypeEngine]:
         return None
 
     @classmethod
-    def apply_limit_to_sql(cls, sql: str, limit: int, database: "Database") -> str:
-        new_sql = ParsedQuery(sql).set_alias()
-        return super().apply_limit_to_sql(new_sql, limit, database)
+    def extract_error_message(cls, ex: Exception) -> str:
+        if str(ex).startswith("(8155"):
```

Comment:
```suggestion
        if str(ex).startswith("(8155,"):
```
You never know if they introduce `^\(8155\d+,`..

Reply:
good point